### PR TITLE
Update zone capacity script update

### PR DESCRIPTION
**Pull request by mathilde-daugy**
## Issue
When running the capacity update, the same value gets added twice which makes the config huge and unreadable for 0 added information
image

## Description

The purpose of this PR is to:
- not update the capacity config if the incoming value is the same as the last one 
- update the date of the capacity point for backfilling to get the earliest date where the value is valid

### Preview

image

### Double check

- [ ] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [ ] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

**Review by unitrium — CHANGES_REQUESTED** [review]
I'm not sure the changes are clear to me, I would recommend improving the code readability a bit and then we can discuss it the two of us.

File: scripts/update_capacity_configuration.py [hunk 1]
```diff
@@ -128,23 +128,64 @@ def generate_zone_capacity_config(
...
 
+def update_capacity_dict_if_value_already_exists(
+    mode: str, capacity_config: dict[str, Any], data: [str, Any]
+) -> list[dict[str, Any]]:
+    """Update the capacity config for a zone if the capacity config is a list and the value already exists. This function ensures that we don't add the same value to the config over and over and that we can backfill and get the oldest value for which the capacity is valid.s"""
+
+    if data[mode]["datetime"] > capacity_config[mode]["datetime"]:
+        # if the associated datetime is more recent than the existing one, we don't add it
+        return capacity_config[mode]
+    else:
```
[the remark this answers is not in the document]
Comment (mathilde-daugy) [reply scripts/update_capacity_configuration.py:140]: my bad, I see that i made a mistake previously with the typing of the function arguments

**Comment by mathilde-daugy** [comment]
@unitrium I've changed the structure of the code so that we check if the value is in the capacity data point rather than the datetime and it seems to make things a bit lighter. I will find some time for us next week so that we can review :)

File: scripts/update_capacity_configuration.py [hunk 2]
```diff
@@ -128,31 +128,85 @@ def generate_zone_capacity_config(
...
+def update_capacity_dict_if_value_already_exists(
+    mode: str, capacity_config: dict[str, Any], new_capacity: dict[str, Any]
+) -> dict[str, Any]:
+    """Updates the capacity config for a zone if the capacity config is a list and the value already exists.
+    This function ensures that we don't add the same value to the config over and over and that we can backfill and get the oldest value for which the capacity is valid.
+    """
+    if new_capacity[mode]["datetime"] > capacity_config[mode]["datetime"]:
+        # if the associated datetime is more recent than the existing one, we don't add it
+        return capacity_config[mode]
+    else:
```
Comment (unitrium) [inline scripts/update_capacity_configuration.py:140]:
```suggestion
```
You can get rid of the else here.

**Review by unitrium — APPROVED** [review]
Looks better! We can still talk about it next week if you want.

**Pull request merged**